### PR TITLE
quaternion, so3 and se3 as non batched

### PR DESCRIPTION
**Pull request by edgarriba**
#### Changes

Improves the `Quaternion`, `So3` and `Se3` api so that one can work with unbatched data keeping the original sophus api.
In addition we made So3 and Se3 torch modules so that we can easily cast data, send to devices or to the optmizers.

#### Type of change

- [x] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [x] 🔬 New feature (non-breaking change which adds functionality)
- [x] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

**Comment by edgarriba** [comment]
@Hackerman342 @stevenlovegrove  /cc @strasdat @ethanrublee

**Comment by stevenlovegrove** [comment]
Excited to try it!

**Review by ducha-aiki — APPROVED** [review]

File: kornia/geometry/liegroup/se3.py [hunk 1]
```diff
@@ -36,17 +39,19 @@ def __init__(self, r: So3, t: Tensor) -> None:
...
+            Parameter containing:
+            tensor([[1., 0., 0., 0.]], requires_grad=True)
             >>> s.t
-            tensor([[1., 1., 1.]])
+            Parameter containing:
+            tensor([[1., 1., 1.]], requires_grad=True)
         """
+        super().__init__()
         KORNIA_CHECK_TYPE(r, So3)
         KORNIA_CHECK_SHAPE(t, ["B", "3"])
```
Comment (cjpurackal) [inline kornia/geometry/liegroup/se3.py:52]: Getting some issues when t is passed as non batched, guess its a bug in KORNIA_CHECK_SHAPE.
```
(Pdb) tt = torch.ones(5)
(Pdb) Se3(So3.identity(1), tt)
rotation: Parameter containing:
tensor([[1., 0., 0., 0.]], requires_grad=True)
translation: Parameter containing:
tensor([1., 1., 1., 1., 1.], requires_grad=True)

```

Comment (edgarriba) [reply kornia/geometry/liegroup/se3.py:52]: Yes, this needs to be addressed separately, for now we need to carefully pass both so3 and translation vector as non batched

**Pull request merged**